### PR TITLE
Fix NotebookEditor initial selected

### LINKED ISSUE #1790: NotebookEditor initial selected
If you have a notebook style list editor, by default, the first tab is shown / selected, however the `selected` trait remains `None` initially.

This happens because `NotebookEditor.init` is called, which connects `currentChanged` qt signal to the method `_tab_activated` which sets the `selected` trait by looking through `_uis`.  Then, `update_editor` gets called, which goes through each item in the list (ie `self.value`), calls `_create_page` and then adds that page to `_uis`.  _However_, that first call to `_create_page` will hit a `self.control.addTab` call which, since before that the QTabWidget was empty, will cause `currentChanged` to fire and `_tab_activated` to be called but this happens _before_ we add this page to `_uis` so `_tab_activated` is effectively a no-op when it shouldn't be.

### PR DESCRIPTION
**Pull request by aaronayres35**
closes #1790 

**Checklist**
- [x] Add a news fragment if this PR is news-worthy for end users. (see docs/releases/README.rst)

### CODE REVIEW

File: traitsui/qt4/list_editor.py
Comment (aaronayres35) [inline traitsui/qt4/list_editor.py:530]: These fixes are a little clunky, but they get the job done.  Maybe it is worth thinking about it more and coming up with something cleaner?  

Goal here was really a quick fix

**Review by rahulporuri — APPROVED** [review]
The code changes LGTM but i'd recommend waiting for @siddhantwahal s approval as well.

File: traitsui/qt4/list_editor.py
Comment (siddhantwahal) [inline traitsui/qt4/list_editor.py:531]: Can `self._tab_activated` be placed outside the for loop? That makes it clearer that we always want the first tab to be selected. 

The tests seem do seem to pass if it's outside.

```suggestion
        self._tab_activated(0)
        if self.selected:
```

**Review by siddhantwahal — APPROVED** [review]
Looks good to me to, one question though:

File: traitsui/qt4/list_editor.py
Comment (aaronayres35) [reply traitsui/qt4/list_editor.py:531]: I think this makes sense / is safe 👍

**Pull request merged**
